tests/unit/replace-widget.spec.ts

```diff
@@ -30,24 +30,46 @@ describe('Widget ReplaceWidget', () => {
       store: emptyStore,
       localVue,
       sync: false,
-      data: () => {
-        return { toReplace: ['foo', 'bar'] };
+      propsData: {
+        value: ['foo', 'bar'],
       },
     });
     const widgetWrappers = wrapper.findAll('inputtextwidget-stub');
     expect(widgetWrappers.at(0).props().value).toEqual('foo');
     expect(widgetWrappers.at(1).props().value).toEqual('bar');
   });
 
-  it('should emit "input" event on "toReplace" update', async () => {
+  it('should emit "input" event with correct updated values when input valueToReplace is updated', async () => {
     const wrapper = shallowMount(ReplaceWidget, {
+      propsData: {
+        value: ['yolo', 'bim'],
+      },
+      store: emptyStore,
+      localVue,
+      sync: false,
+    });
+    await localVue.nextTick();
+    wrapper
+      .findAll('InputTextWidget-stub')
+      .at(0)
+      .vm.$emit('input', 'foo');
+    expect(wrapper.emitted().input[1][0]).toEqual(['foo', 'bim']);
+  });
+
+  it('should emit "input" event with correct updated values when input newValueToReplace is updated', async () => {
+    const wrapper = shallowMount(ReplaceWidget, {
+      propsData: {
+        value: ['yolo', 'bim'],
+      },
       store: emptyStore,
       localVue,
       sync: false,
     });
-    wrapper.setData({ toReplace: ['foo', 'bar'] });
     await localVue.nextTick();
-    expect(wrapper.emitted().input).toBeDefined();
-    expect(wrapper.emitted().input[1]).toEqual([['foo', 'bar']]);
+    wrapper
+      .findAll('InputTextWidget-stub')
+      .at(1)
+      .vm.$emit('input', 'bar');
+    expect(wrapper.emitted().input[1][0]).toEqual(['yolo', 'bar']);
   });
 });
```

Comment:
I think that file misses a test where we don't set any prop (default case), and test the initial default value emission.